### ISSUE TITLE
listview item repeat open，when do noting in fillValues() method

### ISSUE THREAD

**Issue by CGmaybe10** [daimajia/AndroidSwipeLayout#215]
![code](https://cloud.githubusercontent.com/assets/10300510/9560504/e24159ae-4e49-11e5-8b07-e64e98911be5.png)
![first](https://cloud.githubusercontent.com/assets/10300510/9560505/ef294104-4e49-11e5-959f-3a87f129e4b4.png)
![eleventh](https://cloud.githubusercontent.com/assets/10300510/9560507/f3cadbb4-4e49-11e5-9375-20f67b093898.png)

use your example, do nothing in fillValues(int position, View convertView) method, there is a problem. 
eg, listView contains fifty items. 
1step, i open first item,
2step, i slide listView down
when eleventh item shows， It‘s opened！

**Comment by lancewong2014** [comment]
Yeah,I met the same error.It takes me some times to find out why.Sometimes,the listview's item didn't call "onLayout" method！Under this situation,SwipeLayout doesn't call "onLayout",so item is repeated.
I advise to close it manually by a special flag.

**Comment by euangelion666** [comment]
@lancewong2014 could you tell me how to deal with this situation in detail? I met the same error, and I have done something in fillValues() but the item opened as well.

**Comment by lancewong2014** [comment]
@euangelion666 I mean u can add a  flag for each item.When the flag is true,u open it,otherwise u close it.

**Comment by euangelion666** [comment]
@lancewong2014 thank you for your help. But I found that change the view in swipelayout from match_parent to wrap_content can prevent this situation happen. Maybe will help you

**Comment by prashanthd** [comment]
Hi, 
any one found the solution for this problem? I am facing same issue.

**Comment by prashanthd** [comment]
Hi I Know my list is small so used following solution

`@Override

public int getViewTypeCount() {                 

    return getCount();
}

@Override
public int getItemViewType(int position) {

    return position;
}`
